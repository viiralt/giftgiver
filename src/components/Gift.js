import React, { Component } from 'react';
import styled from 'styled-components';

import { Wrapper, Button } from '../elements';

export default class Gift extends Component {
  state = {
    person: '',
    present: '',
  };

  onChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  onSubmit = event => {
    event.preventDefault();
  };

  render() {
    const { person } = this.state;
    return (
      <Wrapper>
        <Form onSubmit={this.onSubmit}>
          <InputPerson
            type="text"
            name="person"
            placeholder="please enter gift recipient"
            onChange={event => this.setState({ person: event.target.value })}
            value={person}
          />
        </Form>
        <Button onClick={this.handleButtonClick} type="submit">
          Submit
        </Button>
      </Wrapper>
    );
  }
}

const Form = styled.form``;
export const InputPerson = styled.input``;
