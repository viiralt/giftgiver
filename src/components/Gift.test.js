import React from 'react';
import { shallow } from 'enzyme';

import Gift from './Gift';
import { InputPerson } from './Gift';

describe('Gift', () => {
  const wrapper = shallow(<Gift />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('initialises person and present in `state`', () => {
    expect(wrapper.state()).toEqual({ person: '', present: '' });
  });

  describe('when typing into the person input', () => {
    const person = 'Uncle Bobo';
    beforeEach(() => {
      wrapper.find(InputPerson).simulate('change', {
        target: { value: person },
      });
    });

    it('updates the person in `state`', () => {
      expect(wrapper.state().person).toEqual(person);
    });
  });
});
