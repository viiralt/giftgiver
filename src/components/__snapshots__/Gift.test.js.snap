// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gift />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.form
          onSubmit={[Function]}
>
          <styled.input
                    name="person"
                    onChange={[Function]}
                    placeholder="please enter gift recipient"
                    type="text"
                    value=""
          />
</styled.form>,
        <styled.button
          onClick={undefined}
          type="submit"
>
          Submit
</styled.button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.input
            name="person"
            onChange={[Function]}
            placeholder="please enter gift recipient"
            type="text"
            value=""
/>,
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "person",
            "onChange": [Function],
            "placeholder": "please enter gift recipient",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Submit",
          "onClick": undefined,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Submit",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.form
            onSubmit={[Function]}
>
            <styled.input
                        name="person"
                        onChange={[Function]}
                        placeholder="please enter gift recipient"
                        type="text"
                        value=""
            />
</styled.form>,
          <styled.button
            onClick={undefined}
            type="submit"
>
            Submit
</styled.button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styled.input
              name="person"
              onChange={[Function]}
              placeholder="please enter gift recipient"
              type="text"
              value=""
/>,
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "person",
              "onChange": [Function],
              "placeholder": "please enter gift recipient",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Submit",
            "onClick": undefined,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
